lib/analyzer/applications/java.ts

```diff
@@ -0,0 +1,58 @@
+import * as path from "path";
+import { JarFingerprint } from "../types";
+import { AppDepsScanResultWithoutTarget, FilePathToContent } from "./types";
+
+function mapContentToJarFingerprints(input: {
```

Comment:
nit: "mapping the content" doesn't convey enough information about what the function is doing so it means the abstraction is gone - you have to go into the code to understand what it is doing. WDYT of `groupJarFingerprintsByPath()` or something like that?